src/components/App.tsx

```diff
@@ -168,6 +168,17 @@ export class App extends React.Component<any, AppState> {
       () => globalSceneState.getAllElements(),
     );
     this.actionManager.registerAll(actions);
+    const url = new URL(document.location.toString()).searchParams.get(
+      "excalidraw",
+    );
+    if (url) {
+      fetch(url, {
+        method: "GET",
+        redirect: "follow",
+      })
+        .then((response) => response.text())
+        .then((result) => this.syncActionResult(JSON.parse(result)));
```

Comment:
This will use the streaming JSON parser in browsers for slightly faster results.

```suggestion
        .then((response) => response.json())
        .then((result) => this.syncActionResult(result));
```

Reply:
Yes, thank you